src/doc/playback.cpp

```diff
@@ -436,13 +437,21 @@ bool Playback::decrementRepeat(const frame_t frameDelta)
                  "repeat=", m_playing.back()->repeat,
                  "forward=", m_playing.back()->forward);
       // Tag has only 1 frame, then don't move the playback cue.
-      return tag->frames() > 1;
+      return tag->frames() > 1 &&
```

Comment:
I've been testing this changes, and they work as expected. But I think this can be simplified by just returning false in this line. Because the only place where the returned value of this function is used is in the following fragment of code in handleExitFrame():

```c++
        else if ((tag->aniDir() == AniDir::PING_PONG ||
                  tag->aniDir() == AniDir::PING_PONG_REVERSE)
                 && m_frame == lastTagFrame(tag)) {
          PLAY_TRACE("    Changing direction frame=", m_frame,
                     " forward=", forward, "->", -forward);

          // Changing the direction of the ping-pong animation
          m_playing.back()->invertForward();  
          return decrementRepeat(frameDelta);    <-------------- here
        }
```
And it actually makes sense, because in `decrementRepeat()`, after decrementing the repeat count, it goes to the first tag frame so we must avoid the moving to the next frame because it will jump the first frame otherwise.

Reply:
Sorry. I tried that solution but it didn't work in all cases. Simple test cases go wrong.
My first approach discarded: https://github.com/aseprite/aseprite/commit/2f72f3e7ec3334f2bd5624dadd49528a81d8d74c

Reply:
No, I meant to return false from decrementRepeat:
```c++
bool Playback::decrementRepeat(const frame_t frameDelta)
{
  bool move = true;
  while (true) {
    Tag* tag = this->tag();
    PLAY_TRACE("    Decrement tag", tag->name(),
               "repeat", m_playing.back()->repeat, "-1");

    if (m_playing.back()->repeat > 1) {
      --m_playing.back()->repeat;
      goToFirstTagFrame(tag);

      PLAY_TRACE("    Repeat tag", tag->name(), " frame=", m_frame,
                 "repeat=", m_playing.back()->repeat,
                 "forward=", m_playing.back()->forward);
      // Tag has only 1 frame, then don't move the playback cue.
      return false;
    }
   . . .
```

This way you are still using the cases where it returns true.